Readme.md

```diff
@@ -1,90 +1,179 @@
-- Hook-up an API Description language Style Guide engine in and execute examples against Rules assertion code as fixtures in your tests (see: `/styleguide/sample-styleguide/linter/linter`
+### 3. Development of the rules
 
-- Make the linter **pass** on all **good examples**
-- Make the linter **fail** on all **bad examples**
+#### 3.1 Install prerequisites
 
+- install [`node.js` v6+](https://nodejs.org/en/download/)
+- install dependencies
+
+```bash
+$ npm install
 ```
-$ ./scripts/test
-Testing styleguide 'sample-styleguide'
-  Testing rule 'sample-rule'
-    Expecting good example 'sample-good-example' to pass
-      PASS
-    Expecting bad example 'sample-bad-example' to fail
-      PASS
+
+#### 3.2 Create new styleguide rule directory structure and files
+
+Fork this [repo](https://github.com/apiaryio/constitutions)
+
+Run:
+
 ```
+$ ./scripts/init name-of-your-new-rule-directory
 
-### 4. Generate the documentation
+```
 
-- Compile the Style Guide Documentation to a Markdown document
+The following structure should be created in `/styleguides` directory
 
 ```
-$ ./scripts/compile
+├ main-rule-dir
+│  └examples                                - examples directory
+│     ├ bad                                 - bad examples directory
+│     │  ├ bad-example1
+│     │  │  ├ api-description-document      - API description document that should fail
+│     │  │  └ meta.yaml                     - configuration and metadata for the example
+│     │  └ bad-example2
+│     │     ├ api-description-document
+│     │     └ meta.yaml
+│     └ good
+│        └ good-example1                    - good examples directory
+│           ├ api-description-document      - API description document that should pass
+│           └ meta.yaml                     - configuration and metadata for the example
+├ functions.js                              - validation function definition
+├ rule.yaml                                 - rule configuration and metadata
+└ README.md                                 - generated readme file (do not edit)
 ```
 
-- [Sample Style Guide Readme](./styleguides/sample-styleguide/README.md)
-- [Sample Rule Readme](./rules/sample-rule/README.md)
-- [Sample Good Example Readme](./rules/sample-rule/good-examples/sample-good-example/README.md)
-- [Sample Bad Example Readme](./rules/sample-rule/bad-examples/sample-bad-example/README.md)
+#### 3.3 Write the examples
 
+Write at least one good and bad API description document.
 
+#### 3.4 Write the validation function
 
-### 5. Collaborate, re-use, share, iterate
+Write the validation function to `functions.js`
 
-- Generate the documentation
-- Not all rules can be expressed in current API Description languages, contrinute to their design
-- Fork this repo, add your Rules, Good and Bad Eaxmples, and Styleguides
+Function must be defined by [declaration](https://www.w3schools.com/js/js_function_definition.asp). 
+Function must be exported by 
+    
+```javascript
+module.exports = {
```

Comment:
Are we saying somewhere how to write the function itself? Like for example what to `return`, how to throw error etc.?

Reply:
https://github.com/apiaryio/constitutions/pull/3/files/10f664ce1c1efd80fd491e2def64ddb623a3e41f#diff-1e290ac8433d555bce009b162cb869d0R83